.editorconfig

```diff
@@ -0,0 +1,4 @@
+﻿[*.cs]
```

Comment:
If we agree on some styles, we could start defining expectations in this file (force empty line at the end of the file, number of tabs/spaces per file type...).

Reply:
We want to use the standard .editorconfig (dotnet new editorconfig), but roslyn/dotnet format cannot fix naming rules for all. I did not look much further into that, but there was no simple solution to apply all rules to all files, unfortunately.

Reply:
A big part of the styling rules now have code fixes associated, might be worth trying again at some point (happy to have a look anyways).

Reply:
They even had when I looked into that last time, it was only 2 weeks ago 😁 

But some of the fixes cannot be applied in bulk (e.g. naming style violation we get for every private field because we don't prefix them with `_` as the default style dictates), this is a limitation of dotnet format / Roslyn as mentioned here: https://github.com/dotnet/format/issues/348#issuecomment-767044552

dotnet format also has trouble formatting our solution. It fails with: 

```
Unhandled exception: System.Exception: Restore operation failed.
   at Microsoft.CodeAnalysis.Tools.CodeFormatter.OpenMSBuildWorkspaceAsync
```
And the fixes end up with a lot of code being commented out in "unmerged changes" blocks, which prevents the codebase from building.

I am now looking into ways of doing the formatting automatically, so we can use the default editor config, and also automatically reformat PRs. Because there won't ever be the right time to merge otherwise :)